geode-core/src/main/java/org/apache/geode/internal/cache/TXState.java

```diff
@@ -1540,6 +1540,11 @@ protected TXEntryState txReadEntry(KeyInfo keyInfo, InternalRegion internalRegio
         txr = txWriteRegion(internalRegion, keyInfo);
       }
       result = dataReg.createReadEntry(txr, keyInfo, createIfAbsent);
+      if (result == null) {
+        assert !createIfAbsent;
```

Comment:
Does "!createIfAbsent" indicate that this is a set operation? If so I think you should add a comment here.
Something like "createReadEntry will only returns null if createIfAbsent is false. CreateIfAbsent will only be false when this method is called by set operations. In that case we do not want the TXState to have a TXEntryState."